### DISCUSSION TITLE
Why all my edges are judged to be boundaries

### DISCUSSION THREAD

**Discussion by Mrchs** [Whinarn/UnityMeshSimplifier#67] [General]
I'm interested in how to keep the boundary of my model and reduce the number of meshes. I don't see how he knows that an edge is a boundary, why all the edges of my model are judged to be boundaries, resulting in no mesh simplification
PS: I changed the original PreserveBorderEdges setting to true`PreserveBorderEdges = true`

**Comment by Whinarn** [comment]
Hi @Mrchs, a border edge is simply an edge that only one triangle is using (that is two vertices only shared by one single triangle).
In most cases PreserveBorderEdges is not desired as it limits the simplification algorithm greatly and the end result isn't close to expectations.

I introduced a feature that I called "smart linking" as described in the readme, which should be enabled by default (if I'm not mistaken). This feature aims to solve problems like holes in meshes where two triangles might share an edge but use duplicated vertices because they want different attributes. If you use large scale meshes you might want to increase the `VertexLinkDistance` based on what works. But be careful of setting this too high, since it will also generate unwanted results. This is simply the maximum allowed distance between two vertices to count as the same in terms of mesh simplification.

It's difficult to help more without knowing more details or seeing the mesh. So I hope this information helps!

**Comment by Mrchs** [comment]
Thank you very much for your reply in your busy schedule. The main problem I had was that even with EnabledSmartLinks, there were large holes in the model. I didn't know what was wrong, and I wanted to try to fix it. At first I thought that PreserveBorderEdges was not turned on, but I found that most of the edges were considered to be boundaries after the PreserveBorderEdges&nbsp;&nbsp;was opened, thus, the number of triangles in my simplified model hardly changed (of course, there is no hole problem)&nbsp;


This is my simplified before and after picture and GLTF format model


I look forward to hearing from you. Thank you again





陈海粟
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Whinarn/UnityMeshSimplifier"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月1日(星期五) 凌晨0:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Whinarn/UnityMeshSimplifier] Why all my edges are judged to be boundaries (Discussion #67)





 
Hi @Mrchs, a border edge is simply an edge that only one triangle is using (that is two vertices only shared by one single triangle).
 In most cases PreserveBorderEdges is not desired as it limits the simplification algorithm greatly and the end result isn't close to expectations.
 
I introduced a feature that I called "smart linking" as described in the readme, which should be enabled by default (if I'm not mistaken). This feature aims to solve problems like holes in meshes where two triangles might share an edge but use duplicated vertices because they want different attributes. If you use large scale meshes you might want to increase the VertexLinkDistance based on what works. But be careful of setting this too high, since it will also generate unwanted results. This is simply the maximum allowed distance between two vertices to count as the same in terms of mesh simplification.
 
It's difficult to help more without knowing more details or seeing the mesh. So I hope this information helps!
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;


从QQ邮箱发来的超大附件

Unnamed-1H-J013.gltf (3.29M, 无限期)进入下载页面：https://mail.qq.com/cgi-bin/ftnExs_download?k=7a653034ac66fbc85928ba28143102184b110451050003031e57560d561c080655571d560002061a015455520704030055045551322530625d0b515957551d067b487a0403021e505f1156340f&t=exs_ftn_download&code=3e042107




Unnamed-1H-J013.gltf.meta (791B, 无限期)进入下载页面：https://mail.qq.com/cgi-bin/ftnExs_download?k=793238348b9d9b9b5d7fb2284461564b4f465a01520000561a560c50034c0505070315055b025c49070400015b58515501560b0062786431595c5959070549557f1f720453524a035b465e1a0f041005370f&t=exs_ftn_download&code=7284badd
